web-app/HeimScripts/heatmap/run.R

```diff
@@ -189,118 +256,140 @@ formatSubset <- function(subsetNumber){
+  totalMembers <- attributes(d)$members
+  add_json <- function(x, start) {
+    members <- attributes(x)$members
+    height <- attributes(x)$height
+    index <- (start - 1):(start + members - 2)
+    index <- paste(index, collapse = ' ')
+    jsonString <<- paste(
+      jsonString,
+      sprintf(
+        '{"height":"%s", "index":"%s", "children":[',
+        height,
+        index
+      )
+    )
+    if (is.leaf(x)) {
+      jsonString <<- paste(jsonString, ']}')
+    } else {
+      add_json(x[[1]], start)
+      jsonString <<- paste(jsonString, ",")
+      leftMembers <- attributes(x[[1]])$members
+      add_json(x[[2]], start + leftMembers)
+      jsonString <<- paste(jsonString, "]}")
     }
-    jsonString <- ""
-    add_json(d, TRUE)
-    return(jsonString)
+  }
+  jsonString <- ""
+  add_json(d, 1)
+  return(jsonString)
 }
 
 
-addClusteringOutput <- function(jsn, measurements){
-
-  colDendrogramEuclideanComplete <- computeDendrogram(t(measurements), 'euclidean', 'complete')
-  colDendrogramEuclideanSingle <- computeDendrogram(t(measurements), 'euclidean', 'single')
-  colDendrogramEuclideanAverage <- computeDendrogram(t(measurements), 'euclidean', 'average')
-  rowDendrogramEuclideanComplete <- computeDendrogram(measurements, 'euclidean', 'complete')
-  rowDendrogramEuclideanSingle <- computeDendrogram(measurements, 'euclidean', 'single')
-  rowDendrogramEuclideanAverage <- computeDendrogram(measurements, 'euclidean', 'average')
-
-  colDendrogramManhattanComplete <- computeDendrogram(t(measurements), 'manhattan', 'complete')
-  colDendrogramManhattanSingle <- computeDendrogram(t(measurements), 'manhattan', 'single')
-  colDendrogramManhattanAverage <- computeDendrogram(t(measurements), 'manhattan', 'average')
-  rowDendrogramManhattanComplete <- computeDendrogram(measurements, 'manhattan', 'complete')
-  rowDendrogramManhattanSingle <- computeDendrogram(measurements, 'manhattan', 'single')
-  rowDendrogramManhattanAverage <- computeDendrogram(measurements, 'manhattan', 'average')
-
+addClusteringOutput <- function(jsn, measurements) {
+  euclideanDistancesRow <- dist(measurements, method = "euclidean")
+  manhattanDistancesRow <- dist(measurements, method = "manhattan")
+  euclideanDistancesCol <- dist(t(measurements), method = "euclidean")
+  manhattanDistancesCol <- dist(t(measurements), method = "manhattan")
+  
+  colDendrogramEuclideanComplete <- computeDendrogram(t(measurements), euclideanDistancesCol, 'complete')
+  colDendrogramEuclideanSingle <- computeDendrogram(t(measurements), euclideanDistancesCol, 'single')
+  colDendrogramEuclideanAverage <- computeDendrogram(t(measurements), euclideanDistancesCol, 'average')
+  rowDendrogramEuclideanComplete <- computeDendrogram(measurements, euclideanDistancesRow, 'complete')
+  rowDendrogramEuclideanSingle <- computeDendrogram(measurements, euclideanDistancesRow, 'single')
+  rowDendrogramEuclideanAverage <- computeDendrogram(measurements, euclideanDistancesRow, 'average')
+  
```

Comment:
No need to pass measurements/t(measurements) to computeDendrogram: argument is not used